src/snowflake/connector/compat.py

```diff
@@ -23,9 +24,11 @@
 IS_MACOS = platform.system() == "Darwin"
 
 NUM_DATA_TYPES: tuple[type, ...] = ()
+numpy: ModuleType | None
 try:
-    import numpy
+    import numpy  # type: ignore
 
+    assert isinstance(numpy, ModuleType)
```

Comment:
may I ask why we add the assertion here? I feel it will always be true

Reply:
This is done to please `mypy` More info here: https://mypy.readthedocs.io/en/stable/type_narrowing.html

Reply:
why we need numpy declaration at line 27?
I feel we can do `# type: ignore` at line 47 to get rid of line 27 and the assertion?